src/storage/table/column_segment.cpp

```diff
@@ -267,7 +267,7 @@ static void FilterSelectionSwitch(T *vec, T *predicate, SelectionVector &sel, id
 }
 
 template <bool IS_NULL>
-static idx_t TemplatedNullSelection(SelectionVector &sel, idx_t approved_tuple_count, ValidityMask &mask) {
+static idx_t TemplatedNullSelection(SelectionVector &sel, idx_t &approved_tuple_count, ValidityMask &mask) {
```

Comment:
Why is this change necessary?

Reply:
Actually this is question #3 I raised in PR description. 

> 3.Even if I pushed the IsNullFilter, The result.size() of DataTable::Scan() seems include all the rows. I traced into the TemplatedNullSelection(), and find the second parameter approved_tuple_count is passed as value, is it expected? Anyway, I changed it to a reference and set it to result size, seems it works for me. Please let me know if I mistake anything.

TemplatedNullSelection is called in
```
idx_t ColumnSegment::FilterSelection(SelectionVector &sel, Vector &result, const TableFilter &filter,
                                     idx_t &approved_tuple_count, ValidityMask &mask) {
```

The reference approved_tuple_count passed in is set by all filters except for IS_NULL and IS_NOT_NULL, is it on purpose?

Reply:
Actually this is question [#3] I raised in PR description.

> 3.Even if I pushed the IsNullFilter, The result.size() of DataTable::Scan() seems include all the rows. I traced into the TemplatedNullSelection(), and find the second parameter approved_tuple_count is passed as value, is it expected? Anyway, I changed it to a reference and set it to result size, seems it works for me. Please let me know if I mistake anything.

I assueme approved_tuple_count is count of qualified(is null) rows, shouldn't it?
TemplatedNullSelection is called in 
```
idx_t ColumnSegment::FilterSelection(SelectionVector &sel, Vector &result, const TableFilter &filter,
                                     idx_t &approved_tuple_count, ValidityMask &mask) {
```

The reference approved_tuple_count passed in is set by all filters except for IS_NULL and IS_NOT_NULL, I changed it and seems it fix my problem. Please let me know if I make a mistake here.